tests/test_roadm_restrictions.py

```diff
@@ -0,0 +1,198 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+# @Author: Esther Le Rouzic
+# @Date:   2019-05-22
+"""
+@author: esther.lerouzic
+checks that fused placed in amp type is correctly converted to a fused element instead of an edfa
+and that no additional amp is added.
+checks that restrictions in roadms are correctly applied during autodesign
+
+"""
+
+from pathlib import Path
+import pytest
+from gnpy.core.utils import lin2db, load_json
+from gnpy.core.elements import Fused, Roadm, Edfa
+from gnpy.core.equipment import load_equipment, Amp, automatic_nch
+from gnpy.core.network import network_from_json, build_network
+
+
+TEST_DIR = Path(__file__).parent
+EQPT_LIBRARY_NAME = TEST_DIR / 'data/eqpt_config.json'
+NETWORK_FILE_NAME = TEST_DIR / 'data/testTopology_expected.json'
+# adding tests to check the roadm restrictions
+
+# mark node_uid amps as fused for testing purpose
+@pytest.mark.parametrize("node_uid", ['east edfa in Lannion_CAS to Stbrieuc'])
+def test_no_amp_feature(node_uid):
```

Comment:
I think that this test is no longer needed; the condition it was designed to test is no longer true. On the other hand, `examples/meshTopologyExampleV2.xls` and `examples/meshTopologyExampleV2.json` now "should be synced". (I'll add a TODO for me to turn this into a test later.)

So I went looking into `tests/data/`, and found `tests/data/testTopology.xls` which was not updated for recent changes. It also appears to have deviated from its likely origin, the `examples/meshTopologyExampleV2.xls`. I went ahead and tried making the requested change, but I think I got stuck somewhere because just running `convert.py` on my modified XLS produces results which I think are wrong.

Can you please also adjust `tests/data/testTopology.xls` so that it includes this new `fused` "amplifier" in that short span, and remove this test?

Reply:
Not sure I understand what you mean by "no more needed" : there is still the possibility that a user needs a roadm with no booster amp, which is what I test here. I have changed the test yesterday: it checks that having a fused element placed just after a roadm disables the automatic addition of an EDFA there. I still think it is needed (to be sure future changes on parsers and build_network do not have regression on this). What is missing is the correct interpretation of fused keyword in parser. I will add that.

Reply:
I would have gone with a different approach where the test contains a predefined JSON file with exactly one `Fused` node at a ROADM egress, but I like this approach as well, and perhaps it's better. Thanks.